events/api.py

```diff
@@ -599,9 +599,16 @@ def __init__(
             # check permissions *before* validation
             if not instance.can_be_edited_by(self.user):
                 raise PermissionDenied()
+
             if isinstance(self.user, ApiKeyUser):
+                # Registration uses event data source for permission check
```

Comment:
The other way for this would be to add a property to `Registration` model:

```python
@property
def data_source(self):
    return instance.event.data_source
```

But, your call. :+1:

Reply:
Yeah, I definitely agree with @charn here, you should do this instead. This class is already making assumptions about what attributes/methods the instances have (e.g. `can_be_edited_by`), which you are implementing in the `Registration` model, so it's a bit of a weird difference.

An even better solution would probably not make assumptions and use interfaces/mixins or something, but that might not be exactly straight-forward.

Reply:
I prefer having data_source property in Registration modal. Checking model type in LinkedEventsSerializer felt a little bit hack to me